Documentation/Blazorise.Docs/Pages/Docs/Extensions/DataGrid/Examples/DataGridLargeDataExpressionCompilerExample.razor

```diff
@@ -0,0 +1,65 @@
+﻿@using Blazorise.DataGrid.Extensions;
+@using Blazorise.DataGrid.Utils
+@namespace Blazorise.Docs.Docs.Examples
+
+<DataGrid @ref=dataGridRef
+          TItem="Employee"
+          Data="@employeeList"
+          ReadData="@OnReadData"
+          TotalItems="@totalEmployees"
+          PageSize="10"
+          ShowPager
+          Responsive
+          Filterable 
+          FilterMode="DataGridFilterMode.Menu">
+    <DataGridCommandColumn />
+    <DataGridColumn Field="@nameof(Employee.FirstName)" Caption="First Name" Editable />
+    <DataGridColumn Field="@nameof(Employee.LastName)" Caption="Last Name" Editable />
+    <DataGridSelectColumn TItem="Employee" Field="@nameof( Employee.Gender )" Caption="Gender" Editable Data="EmployeeData.Genders" ValueField="(x) => ((Gender)x).Code" TextField="(x) => ((Gender)x).Description" />
+    <DataGridNumericColumn Field="@nameof( Employee.Childrens )" Caption="Childrens" Editable />
+    <DataGridDateColumn Field="@nameof( Employee.DateOfBirth )" DisplayFormat="{0:dd.MM.yyyy}" Caption="Date Of Birth" Editable />
+</DataGrid>
+
+@code {
+    [Inject] public EmployeeData EmployeeData { get; set; }
+    private DataGrid<Employee> dataGridRef;
+    private List<Employee> employeeListSource;
+    private List<Employee> employeeList;
+
+    protected override async Task OnInitializedAsync()
+    {
+        employeeListSource = await EmployeeData.GetDataAsync();
+        await base.OnInitializedAsync();
+    }
+
+    private int totalEmployees;
+
+    private async Task OnReadData(DataGridReadDataEventArgs<Employee> e)
+    {
+
+        if (!e.CancellationToken.IsCancellationRequested)
+        {
+            var query = employeeListSource.AsQueryable().ApplyDataGridSort(e.Columns).ApplyDataGridSearch(e.Columns);
```

Comment:
Why do we have `ApplyDataGridSort`, `ApplyDataGridSearch`, and `ApplyDataGridPaging` as extensions? Can't we already have a prebuilt query expression for the ReadData callback argument?

Like `e.Request`?

Reply:
A single method already exists, and you could use it: 
**ApplyDataGridFilters(e.columns);** but the thing is, that you need to calculate the total items. So typically you only want to build part of the query so you can **.Count()**, before you go ahead and apply **Paging**, that's why it's separated into separate methods.

![image](https://github.com/user-attachments/assets/e3f1f79b-7c2a-4b67-b0b3-49889dcdc2f4)


As for the argument you suggest, we could indeed make an additional overload for the **e** => **DataGridReadDataEventArgs**, which would be a little more seamless, but I'd probably keep the **DataGridColumnsInfo** one still, what do you think?

Reply:
> As for the argument you suggest, we could indeed make an additional overload for the e => DataGridReadDataEventArgs, which would be a little more seamless, but I'd probably keep the DataGridColumnsInfo one still, what do you think?

And same thing reasoning applies, where you shouldn't really apply a **paging query before counting**.

Reply:
Also to me **IEnumerable<DataGridColumnInfo>** seems easier to handle if you have to use it on the backend, typically sending it over to an api and de-serializing it, or constructing it manually just to use the feature.

Reply:
> As for the argument you suggest, we could indeed make an additional overload for the **e** => **DataGridReadDataEventArgs**, which would be a little more seamless, but I'd probably keep the **DataGridColumnsInfo** one still, what do you think?

Yes, add it. There are always lazy devs who will appreciate it. They also have extension methods for extra customization. Also, keep the DataGridColumnsInfo .

Reply:
> > As for the argument you suggest, we could indeed make an additional overload for the **e** => **DataGridReadDataEventArgs**, which would be a little more seamless, but I'd probably keep the **DataGridColumnsInfo** one still, what do you think?
> 
> Yes, add it. There are always lazy devs who will appreciate it. They also have extension methods for extra customization. Also, keep the DataGridColumnsInfo .

I mean....
`var query = employeeListSource.AsQueryable().ApplyDataGridSort(e.Columns).ApplyDataGridSearch(e.Columns);`
vs
`var query = employeeListSource.AsQueryable().ApplyDataGridSort(e).ApplyDataGridSearch(e);`

Totally appreciated!! 🤔 hehe

Reply:
Actually, there's a good benefit to your suggestion, this overload here can encapsulate a bit of logic regarding Virtualize vs Paging:
![image](https://github.com/user-attachments/assets/60c1cd48-edc0-4bff-a3e1-a4746d8d520b)